core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -104,13 +90,70 @@ object RequestChannel extends Logging {
       }
     }
 
-    trace(s"Processor $processor received request: ${requestDesc(true)}")
+    trace(s"Processor $processor received request: ${summary.getDescription(true)}")
+
+    val summary: RequestSummary = new RequestSummary(
+      processor,
+      header,
+      metrics,
+      context,
+      session,
+      startTimeNanos,
+      description = s"${context.header} -- ${bodyAndSize.request.toString(false)}",
+      detailedDescription = s"${context.header} -- ${bodyAndSize.request.toString(true)}",
+      sizeInBytes = bodyAndSize.size,
+      isFromFollower = bodyAndSize.request match {
+        case r:FetchRequest => Some(r.isFromFollower)
+        case _ => None
+      }
+    )
+
+    def releaseBuffer(): Unit = {
+      if (buffer != null) {
+        memoryPool.release(buffer)
+        buffer = null
+      }
+    }
+
+    override def toString = s"Request(processor=$processor, " +
+      s"connectionId=${context.connectionId}, " +
+      s"session=$session, " +
+      s"listenerName=${context.listenerName}, " +
+      s"securityProtocol=${context.securityProtocol}, " +
+      s"buffer=$buffer)"
+  }
+
+  /** Summary of Request without the parsed request **/
+  class RequestSummary(
```

Comment:
Maybe include the class of the actual request here?

Reply:
Do you mean include a type reference to the concrete type of the `AbstractRequest`?

Reply:
Yea exactly. Since we're ditching the actual request instance with this change, it might be nice to keep it's Class information around.

Reply:
That's a good idea.  Sure.

Reply:
It looks like the request's type may be redundant with the `RequestHeader.apiKey`, which is used to match requests to handlers in `KafkaApis.handle`.

Reply:
I added the type as a member.